source/index.ts

```diff
@@ -328,6 +328,29 @@ export default class PQueue<QueueType extends Queue<RunFunction, EnqueueOptionsT
 		});
 	}
 
+	/**
+	Wait for the queue size (the number of items not yet started) to fall below a certain threshold.
+
+	@returns A promise that settles when the queue size is below the given limit.
+	*/
+	async onSizeLessThan(limit: number): Promise<void> {
+		// Instantly resolve if the queue is empty.
+		if (this._queue.size < limit) {
+			return;
+		}
+
+		return new Promise<void>(resolve => {
+			const listener = () => {
+				if (this._queue.size < limit) {
+					this.removeListener('next', listener);
+					resolve();
+				}
+			};
+
+			this.on('next', listener);
```

Comment:
Use `.once` instead. Then you don't have to unsubscribe.

Reply:
I do have to keep listening until it reaches the target, though, so if I use `once` I would have to call it again after each event if the target wasn't reached.  Kind of a trade off I guess.  Do you still prefer `once`?

Reply:
You're right. What you have already is a better way to handle it.